### PR TITLE
playground: add test for bad error msg

### PR DESCRIPTION
**Pull request by jchavarri**
Showcases that when warnings are found, the error message passed in the json by the playground app is not good.

### CODE REVIEW

**Comment by jchavarri** [comment]
In the first commit, I added the failing test, the second one adds a fix. cc @davesnx

File: bin/jsoo_main.ml [hunk 1]
```diff
@@ -35,10 +35,34 @@ module Melange_ast = struct
...
+      let msg_of_str str ppf = Format.pp_print_string ppf str in
+      let kind = mk is_error id in
+      let main = { Location.loc; txt = msg_of_str message } in
+      let sub =
+        List.map
+          (fun (loc, sub_message) ->
+            { Location.loc; txt = msg_of_str sub_message })
+          sub_locs
+      in
+      last_warning := Some { Location.kind; main; sub };
```
Comment (anmonteiro) [inline bin/jsoo_main.ml:58]: - can we collect warnings in a list or a queue instead? there might be more than one
- I couldn't find where `last_warning` gets reset? if we compile again, won't we show stale warnings?

Comment (jchavarri) [reply bin/jsoo_main.ml:58]: > can we collect warnings in a list or a queue instead? there might be more than one

I will check how to do this (will add a test to exercize it too).

> I couldn't find where `last_warning` gets reset? if we compile again, won't we show stale warnings?

I was doing it here:

https://github.com/melange-re/melange/blob/5d2731d90dd8baf0500bb1bfc69ea86dccc140c1/bin/jsoo_main.ml#L146

but this is probably not right. E.g. if we got a regular error + a warning, the value of `last_warning` would not be reset, as you mentioned. Will think about this as well.

**Comment by jchavarri** [comment]
@anmonteiro thanks for the review. I adapted the code to handle multiple warnings and reset the warnings collected every time we compile. This modification required extending the interface of the returned JS object to the playground, to add a new type `warning_errors` that includes the array of warnings with the details, I will change playground to support it. cc @davesnx 

Note the branch relies on #654 now, so we should prob merge that PR first before tackling this one (although this one can be reviewed).

**Comment by anmonteiro** [comment]
@jchavarri I think I messed up your branch because I didn't notice it was opened against the other PR.

**Comment by anmonteiro** [comment]
Alright, I fixed this + reversed the list of warnings such that they're shown in the correct order

**Pull request merged**